### ISSUE TITLE
List index (0) out of bounds

### ISSUE THREAD

**Issue by ngraham20** [benjamimgois/goverlay#135]
When starting Goverlay, I always get an error message saying
```
List index (0) out of bounds.

Press OK to ignore and risk data corruption.
Press Abort to kill the program
```

Pressing OK does not *appear* to cause any issues, however I don't know for sure

**Comment by ngraham20** [comment]
System:

Distro: Bazzite 40
Desktop Environment: KDE 6
Installed on Distrobox Fedora toolbox 40 with `sudo dnf install goverlay`

**Comment by benjamimgois** [comment]
Hi @ngraham20 , i have never tried it on Bazzite or atomic desktops. I'll take a look

**Comment by ngraham20** [comment]
It might have something to do with the custom preset. When I try to modify the custom preset, it fails to load all of the settings. By that, I mean if I click `FPS only`, mangohud correctly updates, and if I go through `Visual`, `Performance`, `Metrics`, and `Extras`, all the settings are set for `FPS only`. However, if I click the `Custom` preset, nothing changes. All tabs are still set to the `FPS only` settings.

**Comment by sunnyayyl** [comment]
Coincidentally, I had the same issue on Fedora silverblue (_which is also a atomic desktop?_), and I am running goverlay on a fedora container with distrobox.

**Comment by ngraham20** [comment]
> Coincidentally, I had the same issue on Fedora silverblue (_which is also a atomic desktop?_), and I am running goverlay on a fedora container with distrobox. 

Yeah it's an atomic desktop. And actually Bazzite is based on Silverblue if I understand right

**Comment by benjamimgois** [comment]
> It might have something to do with the custom preset. When I try to modify the custom preset, it fails to load all of the settings. By that, I mean if I click `FPS only`, mangohud correctly updates, and if I go through `Visual`, `Performance`, `Metrics`, and `Extras`, all the settings are set for `FPS only`. However, if I click the `Custom` preset, nothing changes. All tabs are still set to the `FPS only` settings.

@ngraham20 have you saved your custom preset before trying to load it ?

![image](https://github.com/user-attachments/assets/e8b01e2f-979a-4929-b329-abaa3ba70585)

**Comment by ngraham20** [comment]
> > It might have something to do with the custom preset. When I try to modify the custom preset, it fails to load all of the settings. By that, I mean if I click `FPS only`, mangohud correctly updates, and if I go through `Visual`, `Performance`, `Metrics`, and `Extras`, all the settings are set for `FPS only`. However, if I click the `Custom` preset, nothing changes. All tabs are still set to the `FPS only` settings.
> 
> @ngraham20 have you saved your custom preset before trying to load it ?
> 
> ![image](https://github.com/user-attachments/assets/e8b01e2f-979a-4929-b329-abaa3ba70585)
> 

So it does load it correctly, but the actual settings don't load. As in the actual mangohud updates, but the goverlay settings don't update for me to modify

**Comment by SuperSandro2000** [comment]
This bug happens for me whenever I am trying to open the program and old settings are never loaded. That makes it pretty much unusable.

**Comment by Nichta** [comment]
Just happened on a fresh install of NixOS 25.05, with `goverlay 1.3-2` from the nixos repository.

- added `pkgs.goverlay` to `environment.systemPackages`

- performed a `rebuild` and reset the computer

- goverlay shows `list index (0) out of bounds` when ran


![Image](https://github.com/user-attachments/assets/f9c0f06e-e36c-4b50-8b35-8270f46aeba8)

**Comment by furdiburd** [comment]
<img width="1429" height="726" alt="Image" src="https://github.com/user-attachments/assets/621207da-bf8a-427d-9d18-89e5cef5f20a" /> 

Same here, Nixos 25.05, updated to newest version and still the same error, for the time being i am going to use mangojuice

**Comment by benjamimgois** [comment]
@furdiburd  and @Nichta  Are you using the Appimage version ? or the native package ?

**Comment by furdiburd** [comment]
> [@furdiburd](https://github.com/furdiburd) and [@Nichta](https://github.com/Nichta) Are you using the Appimage version ? or the native package ?

I use telhe native package

**Comment by FluffyPuppyKasey** [comment]
This is happening to me on Arch. Native package

**Comment by benjamimgois** [comment]
I just made some fixes to function that locates system fonts, also add some especific fixes for nixOS. @FluffyPuppyKasey and @furdiburd can you try this build ?

[goverlay_nix_fix.tar.gz](https://github.com/user-attachments/files/23659558/goverlay_nix_fix.tar.gz)

**Comment by benjamimgois** [comment]
Fixed in 1.6.2

https://github.com/benjamimgois/goverlay/releases/tag/1.6.2
